lib/screens/v2/widget_instance/alert.ex

```diff
@@ -2,85 +2,210 @@ defmodule Screens.V2.WidgetInstance.Alert do
+      equal?(informed_zones_set, new([:home_stop])) -> :inside
+      equal?(informed_zones_set, new([:upstream, :home_stop, :downstream])) -> :inside
+      equal?(informed_zones_set, new([:upstream, :home_stop])) -> :boundary_upstream
+      equal?(informed_zones_set, new([:home_stop, :downstream])) -> :boundary_downstream
+      true -> :elsewhere
+    end
   end
 
-  def slot_names(%T{}), do: ~w[medium_left medium_right]a
+  @spec effect(t()) :: Alert.effect()
+  def effect(%__MODULE__{alert: %Alert{effect: effect}}), do: effect
 
-  def widget_type(%T{}), do: :alert
+  @spec informed_entity_to_zone(Alert.informed_entity(), map()) ::
+          list(:upstream | :home_stop | :downstream)
+  defp informed_entity_to_zone(informed_entity, location_context)
 
-  defp active_priority(%T{alert: alert}, alert_active?) do
-    if alert_active?.(alert), do: 0, else: 1
+  # All values nil
+  defp informed_entity_to_zone(%{stop: nil, route: nil, route_type: nil}, _location_context) do
+    []
   end
 
-  defp informed_entity_priority(%T{
-         screen: :ok,
-         alert: %Alert{informed_entities: _informed_entities}
-       }) do
-    0
+  # Only route type is not nil--this is the only time we consider route type,
+  # since it's implied by other values when they are not nil
+  defp informed_entity_to_zone(%{stop: nil, route: nil, route_type: route_type_id}, context) do
+    route_type = RouteType.from_id(route_type_id)
+
+    case {route_type, context.screen.app_id} do
+      {:bus, :bus_shelter_v2} -> [:upstream, :home_stop, :downstream]
+      {:bus, :bus_eink_v2} -> [:upstream, :home_stop, :downstream]
+      {:light_rail, :gl_eink_v2} -> [:upstream, :home_stop, :downstream]
+      _ -> []
+    end
   end
 
-  for {effect, priority} <- @effect_priorities do
-    defp effect_priority(%T{alert: %Alert{effect: unquote(effect)}}) do
-      unquote(priority)
+  # Only stop is not nil (route type ignored)
+  defp informed_entity_to_zone(%{stop: stop, route: nil}, context) do
+    cond do
+      stop == context.home_stop -> [:home_stop]
+      stop in context.upstream_stops -> [:upstream]
+      stop in context.downstream_stops -> [:downstream]
+      true -> []
     end
   end
 
-  defp effect_priority(_), do: 100
+  # Only route is not nil (route type ignored)
+  defp informed_entity_to_zone(%{stop: nil, route: route}, context) do
+    # Should we consider whether the stop is a terminus?
+    # If so, an alert that informs a route through the stop might not always be all 3 zones.
+    if route in context.active_routes, do: [:upstream, :home_stop, :downstream], else: []
+  end
 
-  defp serialize_text(%T{alert: _alert}) do
-    ["Dummy alert text"]
+  # Both stop and route are not nil (route type ignored)
+  defp informed_entity_to_zone(%{stop: _stop, route: route} = informed_entity, context) do
+    if route in context.active_routes do
+      informed_entity_to_zone(%{informed_entity | route: nil}, context)
+    else
+      []
```

Comment:
It might be worth checking with Laina + design about checking whether the route is in `active_routes`. It makes sense that we'd want to de-prioritize routes which aren't running on the given day, but I don't think we ever explicitly discussed that case. (Same comment applies to line 168 above.)